test/e2e/test-document-height.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+describes.endtoend(
+  'documentHeight',
+  {
+    testUrl:
+      'http://localhost:8000/test/fixtures/e2e/amp-carousel/0.1/document-height.html',
+    environments: ['viewer-demo'],
+  },
+  async (env) => {
+    let controller;
+
+    beforeEach(async () => {
+      controller = env.controller;
+    });
+
+    it('should send documentHeight once amp has completed init', async () => {
+      const messages = await controller.evaluate(() => {
+        // The viewer.html file will launch 7 test viewers, only one of which is the requested url.
```

Comment:
Might speed up the test if we avoid loading the 6 hard-coded viewers. Issue is we're reusing a manual test page as an e2e test fixture, but a special-case when a custom URL is set would work.

A nice fix-it task perhaps, would you mind filing an issue?

Reply:
https://github.com/ampproject/amphtml/issues/28200